go/teams/hidden/ratchet.go

```diff
@@ -264,24 +264,24 @@ func generateRatchet(mctx libkb.MetaContext, b sig3.Tail) (ret *Ratchet, err err
 	return ret, nil
 }
 
-// MakeRatchet constructs a new Ratachet object for the given team's hidden tail, blinds
+// MakeRatchet constructs a new Ratchet object for the given team's hidden tail, blinds
 // it with a randomly-generated blinding key, and then packages all relevant info up into
 // and encoding that can be easily posted to the API server.
-func MakeRatchet(mctx libkb.MetaContext, id keybase1.TeamID) (ret *Ratchet, err error) {
-
-	defer mctx.Trace(fmt.Sprintf("MakeRatchet(%s)", id), func() error { return err })()
+func MakeRatchet(mctx libkb.MetaContext, state *keybase1.HiddenTeamChain) (ret *Ratchet, err error) {
+	if state == nil {
+		mctx.Debug("hidden.MakeRatchet: returning a nil ratchet since hidden team is nil")
+		return nil, nil
```

Comment:
I'm not a fan of `MakeRatchet -> nil, nil`. Can the caller manage nil?

Reply:
yes, this didn't change at all.  it's not an error case, since you get there if the feature is flagged off.